src/iso15118/message/service_selection.cpp

```diff
@@ -17,6 +17,7 @@ template <> void convert(const struct iso20_ServiceSelectionReqType& in, Service
     out.selected_energy_transfer_service.parameter_set_id = in.SelectedEnergyTransferService.ParameterSetID;
 
     if (in.SelectedVASList_isUsed == true) {
+        out.selected_vas_list.emplace(datatypes::SelectedServiceList({}));
         out.selected_vas_list.value().reserve(in.SelectedVASList.SelectedService.arrayLen);
         auto& vas_list_out = *out.selected_vas_list;
```

Comment:
An alternative:
```
auto& vas_list_out = out.selected_vas_list.emplace();
vas_list_out.reserve(in.SelectedVASList.SelectedService.arrayLen);
```

Reply:
Nice. Let me see if I find can time to reproduce this with your alternative fix. (The linked issue, or one of its linked issues, had a reproducer.)

Reply:
I added my suggestion and some unit tests to test the changes